### ISSUE TITLE
Filenames not included in json output

### ISSUE THREAD

**Issue by drewmullen** [terraform-linters/tflint#1317]
notice filename is empty:

```bash
$ tflint --only=terraform_required_version  -f json | jq
{
  "issues": [
    {
      "rule": {
        "name": "terraform_required_version",
        "severity": "warning",
        "link": "https://github.com/terraform-linters/tflint/blob/v0.34.0/docs/rules/terraform_required_version.md"
      },
      "message": "terraform \"required_version\" attribute is required",
      "range": {
        "filename": "",
        "start": {
          "line": 0,
          "column": 0
        },
        "end": {
          "line": 0,
          "column": 0
        }
      },
      "callers": []
    }
  ],
  "errors": []
}
```

versions.tf
```hcl
terraform {
  # required_version = ">= 0.15.0"
  experiments      = [module_variable_optional_attrs]
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.68"
    }
  }
}
```

## Version

```console
$ tflint -v
TFLint version 0.34.0
+ ruleset.aws (0.12.0)
$ tf  -v
Terraform v1.1.2
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.73.0
```

**Comment by bendrucker** [comment]
There is extensive test coverage demonstrating that this is not related to `--format json`

This is perfectly valid:

```tf
terraform {
  experiments = [module_variable_optional_attrs]
}

terraform {
  required_version = ">= 0.15.0"
  
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.68"
    }
  }
}
```

The high level version constraint representation the rule sees does not include information on the parent `terraform` block where it was defined:

https://github.com/terraform-linters/tflint/blob/a8b001c820d3ac813b153c61f936419e1a45411c/rules/terraformrules/terraform_required_version.go#L48

In theory it's possible to look for existing `terraform` blocks and suggest one of those, but it's strictly correct for the rule to simply say "there's a problem but I don't know which file/location."
